aliyun/src/main/java/org/apache/iceberg/aliyun/AliyunProperties.java

```diff
@@ -25,21 +25,76 @@
 import org.apache.iceberg.util.PropertyUtil;
 
 public class AliyunProperties implements Serializable {
+  /**
+   * The domain name used to access OSS. OSS uses HTTP Restful APIs to provide services. Different regions are accessed
+   * by using different endpoints. For the same region, access over the internal network or over the Internet also uses
+   * different endpoints. For more information, see:
+   * https://www.alibabacloud.com/help/doc-detail/31837.htm
+   */
+  public static final String OSS_ENDPOINT = "oss.endpoint";
+
+  /**
+   * Aliyun uses an AccessKey pair, which includes an AccessKey ID and an AccessKey secret to implement symmetric
+   * encryption and verify the identity of a requester. The AccessKey ID is used to identify a user.
+   * <p>
+   * For more information about how to obtain an AccessKey pair, see:
+   * https://www.alibabacloud.com/help/doc-detail/53045.htm
+   */
+  public static final String ACCESS_KEY_ID = "access.key.id";
+
+  /**
+   * Aliyun uses an AccessKey pair, which includes an AccessKey ID and an AccessKey secret to implement symmetric
+   * encryption and verify the identity of a requester.  The AccessKey secret is used to encrypt and verify the
+   * signature string.
+   * <p>
+   * For more information about how to obtain an AccessKey pair, see:
+   * https://www.alibabacloud.com/help/doc-detail/53045.htm
+   */
+  public static final String ACCESS_KEY_SECRET = "access.key.secret";
+
+  /**
+   * The implementation class of {@link AliyunClientFactory} to customize Aliyun client configurations.
+   * If set, all Aliyun clients will be initialized by the specified factory.
+   * If not set, {@link DefaultAliyunClientFactory} is used as default factory.
+   */
+  public static final String CLIENT_FACTORY = "client.factory-impl";
```

Comment:
would it be better to make this the same as https://github.com/apache/iceberg/blob/master/aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java#L145?

Reply:
It would be nice if they were the same, but I like that this conforms to the `-impl` convention that we use for dynamically loaded classes elsewhere.

Reply:
I can make another PR to line up the behavior of aws part as "client.factory-impl"

Reply:
Cool, that works for me as well